server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -221,6 +221,26 @@ func (r *Runner) Apply(ctx workflow.Context, root *terraform.LocalRoot, serverUR
 func (r *Runner) Run(ctx workflow.Context) error {
 	r.MetricsHandler.Gauge(ActiveTerraformWorkflowStat).Update(1)
 	defer r.MetricsHandler.Gauge(ActiveTerraformWorkflowStat).Update(0)
+	var err error
+	// make sure we are updating state on completion.
+	defer func() {
+		reason := state.SuccessfulCompletionReason
+		if r := recover(); r != nil || err != nil {
+			reason = state.InternalServiceError
```

Comment:
If I'm understanding correctly, if we see any error in the workflow run (`PlanRejectedError` for eg), we set the reason to `state.InternalServiceError`?

Reply:
In the future we'd want to show custom messages based on the error, but given the time crunch right now I'd rather just fix the problem at hand and move on right now.

Reply:
Sounds good to me! Just wanted to make sure